tls/s2n_cipher_preferences.c

```diff
@@ -748,13 +717,14 @@ const struct s2n_cipher_preferences cipher_preferences_kms_fips_tls_1_2_2018_10
         .count = sizeof(cipher_suites_kms_fips_tls_1_2_2018_10) / sizeof(cipher_suites_kms_fips_tls_1_2_2018_10[0]),
         .suites = cipher_suites_kms_fips_tls_1_2_2018_10,
         .minimum_protocol_version = S2N_TLS12,
-        .extension_flag = S2N_ECC_EXTENSION_ENABLED
 
 };
 
 struct {
     const char *version;
     const struct s2n_cipher_preferences *preferences;
+    uint8_t ecc_extension_required;
```

Comment:
Would using a bit field here make sense? I didn't want to combine the two values and do the masking to check values since that seemed more error prone.

Reply:
> Would using a bit field here make sense? I didn't want to combine the two values and do the masking to check values since that seemed more error prone.

Since ecc_extension_required  and pq_kem_extension_required are flags are flags, we could try unsigned var:1 https://github.com/awslabs/s2n/blob/d7366bd22fe21e22b342e81b17175050769ec02c/tls/s2n_connection.h#L74